### ISSUE TITLE
Merge reference documentation of MathCore and MathMore

### ISSUE THREAD

**Issue by ynikitenko** [root-project/root#7440]
### Explain what you would like to see improved
When I browse math documentation, I read MathMore documentation at https://root.cern/doc/master/group__MathMore.html .

When I browse menu in the left column, there are four submenus. 
The submenu `Numerical Algorithms` has on its page https://root.cern/doc/master/group__NumAlgo.html (see that the link is common for both MathCore and MathMore):
Math » MathCore | Math » MathMore
Same for `Statistical functions`.

When I press `Special functions` on the left menu, the left menu changes to MathCore!
https://root.cern/doc/master/group__SpecFunc.html
Same for `QuasiRandom number generators and distributions`.

This is very confusing. 

### Optional: share how it could be improved
I think that MathMore subsections should be deleted (or merged with MathCore, which is larger). 
There can remain the main page describing what MathMore is, but all subsections common with MathCore should belong only there (or to a common ROOT::Math section).

This is very confusing for new users, because I didn't notice that the link is the same, and had to browse both sections in MathCore and MathMore trying to find the difference.

### Additional context
I think this is not related to the site, but to documentation generation. Even if it's not simple to be implemented technically, this is still a feature request (or even a bug report), because for users this is not convenient.

**Comment by couet** [comment]
@lmoneta we need you input there. I already saw that "Numerical Algo" appears in both. But it was like that already before I did the changes. Do we need really need num-algo in both ?

**Comment by couet** [comment]
@lmoneta it is not that we have it "now", it has always been like that. I fully agree that two group like "NumAlgoCore" and "NumAlgoMore" would be much better. Like:

```
// the next classe should be in NumAlgoCore ?
math/mathcore/inc/Math/DistSamplerOptions.h:    @ingroup NumAlgo
math/mathcore/inc/Math/GenAlgoOptions.h:    @ingroup NumAlgo
math/mathcore/inc/Math/IMinimizer1D.h:   @ingroup NumAlgo
math/mathcore/inc/Math/IOptions.h:    @ingroup NumAlgo
math/mathcore/inc/Math/Integrator.h:@defgroup NumAlgo Numerical Algorithms
math/mathcore/inc/Math/Integrator.h:@ingroup  NumAlgo
math/mathcore/inc/Math/Minimizer.h:   @ingroup NumAlgo
math/mathcore/inc/Math/RichardsonDerivator.h:    @ingroup NumAlgo
math/mathcore/inc/Math/RootFinder.h:   @ingroup NumAlgo

// the next classe should be in NumAlgoMore ?
math/mathmore/inc/Math/ChebyshevApprox.h:   @ingroup NumAlgo
math/mathmore/inc/Math/GSLMultiRootFinder.h:       @ingroup NumAlgo
math/mathmore/inc/Math/Interpolator.h:@ingroup NumAlgo
```

I can do it but then you should provide me the little headers  for NumAlgoCore and NumAlgoMore

**Comment by lmoneta** [comment]
@couet , yes we have it now but I am aware it was always like this. The idea was to have the all the classes visible in one place. 
We can have now : 
```
@defgroup NumAlgoCore Numerical Algorithms
@ingroup MathCore
```
and
```
@defgroup NumAlgoMore Numerical Algorithms
@ingroup MathMore
```
I can provide these changes in the doc.

**Comment by couet** [comment]
Yes that's the way. Then change all the classes accordingly.

**Comment by ynikitenko** [comment]
I agree that it would be best for users to have all functions in one place. Menu access to that place should be good and clear though.

**Comment by ynikitenko** [comment]
I faced another problem.
From https://root.cern.ch/doc/master/mathmoreIntegration_8C.html I copy

    #include "Math/Integrator.h"
    ...
    ROOT::Math::Integrator ig(f1 );

but I don't know what the class "Integrator" is. In the documentation it is absent!

https://root.cern.ch/doc/master/group__Integration.html

I think it should either be added to the documentation or removed from ROOT and tutorials if it's obsolete.

O, thanks to the web documentation, I see in the tutorial that it points to https://root.cern.ch/doc/master/classROOT_1_1Math_1_1IntegratorOneDim.html . However, I initially used tutorials from my ROOT package, and they gave no clue where to look for the Integrator.

**Comment by couet** [comment]
I let @lmoneta comment. I can help to modify the doc if needed.

**Comment by ynikitenko** [comment]
Thank you. And one more point.
The include directives

    #include "Math/Integrator.h"

or the namespace

    Math::

give absolutely no hint where one should look for the class Integrator (or another one from that namespace). Is it MathCore, MathMore or what? Maybe it would be better to maintain the docs also in accord with code.

**Comment by hageboeck** [comment]
I had a look and I agree that the topics should be merged. I'm moving them under MathCore now, and I'll add an extra reference to MathMore that will be a subtopic of MathCore, explaining why it exists at all. Ideally, GSL is installed, and the user wouldn't even notice whether they are calling a MathCore or MathMore function.
Locally, I can already see it making more sense, but we need to merge to see the impact on the full ROOT Reference guide.

> Thank you. And one more point. The include directives
> 
> ```
> #include "Math/Integrator.h"
> ```
> 
> or the namespace
> 
> ```
> Math::
> ```
> 
> give absolutely no hint where one should look for the class Integrator (or another one from that namespace). Is it MathCore, MathMore or what? Maybe it would be better to maintain the docs also in accord with code.

Since you opened this issue, it might have gotten better, since it's findable in doxygen. The integrator is a typedef in `ROOT::Math`, so it is findable in Doxygen as the third line of this search dialogue shows:
<img width="335" height="105" alt="Image" src="https://github.com/user-attachments/assets/ed6c20f6-5572-4ad7-a5f5-25c0cdabe5f2" />

Doxygen lists it under ROOT::Math (as it should), so if you unfold the first item in the list, it unfortunately won't be visible there. But this I'm afraid is a consequence of how doxygen searches, so I guess we are limited to what we have now.
